src/DetectChanges/BCBreak/FunctionBased/ReturnTypeCovarianceChanged.php

```diff
@@ -55,7 +62,17 @@ private function typeToString(?ReflectionType $type): string
             return 'no type';
         }
 
-        return ($type->allowsNull() ? '?' : '')
-            . $type->__toString();
+        return $type->__toString();
+    }
+
+    /** @TODO may the gods of BC compliance be merciful on me */
```

Comment:
Similar to above: this is currently the "cleanest" approach, but it highlights the limitation of the current `ReflectionType` design.